### PR TITLE
Optimize startup time, ignore load unConfigured table's column and index

### LINKED ISSUE #6212: Optimize startup time， control which tables can load when startup
I find the startup time is too slow due to the database full tables metadata scan,  
for our case. the database has 200 tables, we are using micro service, one app loads about 10-20 tables,  it's not acceptable to load all tables on startup

tuohai666:

We have received several similar reports about the slow startup recently, mostly due to loading unconfigured tables for meta data purpose of ShardingSphere.

https://github.com/apache/shardingsphere/issues/6179 : 7000+ tables cost much time.
Similar issues: https://github.com/apache/shardingsphere/issues/6161 , https://github.com/apache/shardingsphere/issues/5755 , https://github.com/apache/shardingsphere/issues/5999 .

Loading table (column and index information) is expensive, there're 7 SQLs interactive for each table.
image

After investigate, I found that for tables unconfigured will not use column or index information any more after loading. So we can omit this process. 

I also checked the relevant issue https://github.com/apache/shardingsphere/issues/4460 , which will load collation for all columns in every table. If a table is unconfigured, the route result must be single route, a single route will not need merge any more. If the resultset does not need to merge, it is irrelevant with cache of collation. So I think we can safely remove the loading of unconfigured tables now.

Note that all unconfigured table names still need to be loaded for routing logic, this can simply obtain from one SQL for each database.

We can accomplish the goal by several steps.

- [ ] Remove the load procedure for unconfigured tables.
- [ ] Remove the refresh procedure for unconfigured tables
- [ ] Remove the multi-thread logic for unconfigured tables. (Configured tables still use multi-thread)
- [ ] Load table names for unconfigured tables.
- [ ] Sync up the new load logic with orchestration module.
- [ ] Update the unit test cases.

### PR DESCRIPTION
**Pull request by xbkaishui**
Fixes #6212 .

Changes proposed in this pull request:
- Remove the load procedure for unconfigured tables. load only table names for unconfigured tables.
- Remove the refresh procedure for unconfigured tables
- Remove the multi-thread logic for unconfigured tables. (Configured tables still use multi-thread)

### CODE REVIEW

**Comment by xbkaishui** [comment]
@tristaZero  Please help to review, Thanks

File: shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/engine/dql/BaseDQLIT.java [hunk 1]
```diff
@@ -123,6 +123,9 @@ private void assertMetaData(final ResultSetMetaData actualMetaData, final List<D
         if ("shadow".equals(getRuleType())) {
             return;
         }
+        if (0 == actualMetaData.getColumnCount()) {
```
Comment (tristaZero) [inline shardingsphere-integration-test/shardingsphere-test-suite/src/test/java/org/apache/shardingsphere/dbtest/engine/dql/BaseDQLIT.java:126]: // TODO comment is required.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java [hunk 2]
```diff
@@ -87,22 +78,11 @@ public static SchemaMetaData load(final DataSource dataSource, final int maxConn
         if (tableNames.isEmpty()) {
             return new SchemaMetaData(Collections.emptyMap());
         }
-        List<List<String>> tableGroups = Lists.partition(tableNames, Math.max(tableNames.size() / maxConnectionCount, 1));
-        Map<String, TableMetaData> tableMetaDataMap = 1 == tableGroups.size()
-                ? load(dataSource.getConnection(), tableGroups.get(0), databaseType) : asyncLoad(dataSource, maxConnectionCount, tableNames, tableGroups, databaseType);
+        Map<String, TableMetaData> tableMetaDataMap = new HashMap<>(tableNames.size(), 1);
+        tableNames.forEach(tableName -> tableMetaDataMap.put(tableName, new TableMetaData(Collections.emptyList(), Collections.emptyList())));
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/schema/SchemaMetaDataLoader.java:82]: // TODO comment is required.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/table/TableMetaDataLoader.java [hunk 3]
```diff
@@ -54,6 +55,24 @@
...
+     * @param databaseType database type
+     * @return table meta data
+     * @throws SQLException SQL exception
+     */
+    public static Optional<TableMetaData> loadWithoutColumnMetaData(final DataSource dataSource, final String table, final String databaseType) throws SQLException {
+        try (MetaDataConnection connection = new MetaDataConnection(dataSource.getConnection())) {
+            if (!isTableExist(connection, table, databaseType)) {
+                return Optional.empty();
+            }
+            return Optional.of(new TableMetaData(Collections.emptyList(), Collections.emptyList()));
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-binder/src/main/java/org/apache/shardingsphere/sql/parser/binder/metadata/table/TableMetaDataLoader.java:72]: // TODO comment is required.

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/refresh/impl/CreateTableStatementMetaDataRefreshStrategy.java [hunk 4]
```diff
@@ -53,7 +53,7 @@ public void refreshMetaData(final ShardingSphereMetaData metaData, final Databas
     private void refreshUnconfiguredMetaData(final ShardingSphereMetaData metaData, 
                                              final DatabaseType databaseType, final Map<String, DataSource> dataSourceMap, final String tableName) throws SQLException {
         for (Entry<String, DataSource> entry : dataSourceMap.entrySet()) {
-            Optional<TableMetaData> tableMetaData = TableMetaDataLoader.load(entry.getValue(), tableName, databaseType.getName());
+            Optional<TableMetaData> tableMetaData = TableMetaDataLoader.loadWithoutColumnMetaData(entry.getValue(), tableName, databaseType.getName());
```
Comment (tristaZero) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/metadata/refresh/impl/CreateTableStatementMetaDataRefreshStrategy.java:56]: // TODO comment is required.

**Review by tristaZero — CHANGES_REQUESTED** [review]

**Review by tristaZero — APPROVED** [review]

**Pull request merged**